langlib/lang.array/src/main/ballerina/array.bal

```diff
@@ -70,6 +85,10 @@ public isolated function enumerate(Type[] arr) returns [int, Type][] = @java:Met
 
 # Applies a function to each member of an array and returns an array of the results.
 #
+# ```
+# [0, 1, 2].map(n => n * 2) => [0,2,4]
```

Comment:
In the original PR [we had another example](https://github.com/ballerina-platform/ballerina-lang/pull/38892/files#diff-2c948d23bf7d2faa264a43f44ca32e7f64f06b4dfb220764247218037b2d10f4R91) with an explicit anonymous function expression as the argument.

```
# Employee[] employees = [
#     {firstName: "Jo", lastName: "David", id: 2121},
#     {firstName: "Emma", id: 2122},
#     {firstName: "John", lastName: "Doe", id: 2123}
# ];
# employees.'map(isolated function (Employee employee) returns string {
#     string? lastName = employee.lastName;
# 
#     if lastName is () {
#         return employee.firstName;
#     }
# 
#     return string `${employee.firstName} ${lastName}`;
# }) => ["Jo David","Emma","John Doe"]
# 
# type Employee record {
#     string firstName;
#     string lastName?;
#     int id;
# };
```

Should we continue to have this example too (after the simple one)? IMO, it will be useful to the user.

Applies to some other functions also.

Reply:
I think this is too much to include in a lang doc. A better place could be a BBE.

Reply:
Hmm, I feel we should have examples with above option as well. That really more understandable for a new user.

Reply:
Conclusion was to not add. We're trying to explain the function rather than show the user how to write code.